### PR TITLE
Add WSLtty from https://github.com/mintty/wsltty

### PR DESCRIPTION
**Pull request by me-and**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/denelon/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

[WSLtty](https://github.com/mintty/wsltty) is an open source terminal for connecting to Windows Subsystem for Linux, based on Mintty (the terminal used for Cygwin, and with a few modifications, for Git for Windows), which is in turn based on PuTTY.

(For the sake of explicitness: I am a Microsoft employee, but this project is well outside the scope of my normal work. I don't have any more context/experience/knowledge about winget than anyone else contributing to this project in their own time.)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/40573)

### CODE REVIEW

**Comment by me-and** [comment]
I can't see what's causing the validation checks to fail. I do note that the warning about a lack of quiet flags seems spurious, as the installer.yaml file absolutely includes `InstallerSwitches: Silent: /Q`; my best guess is that that's warning about that flag not being present in a different YAML file and the warning's a hangover from when there was a single YAML file rather than a set.

**Comment by OfficialEsco** [comment]
VirtualBoxVM_296uYQ109R

**Review by ItzLevvie — CHANGES_REQUESTED** [review]
I'm seeing this error:
image

When I run the installer manually:
image

Perhaps this might be something to report to the developers of wsltty? cc @mintty

File: manifests/m/Mintty/WSLtty/3.5.1/Mintty.WSLtty.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Created using wingetcreate 0.4.4.1
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.0.0.schema.json
+
+PackageIdentifier: Mintty.WSLtty
+PackageVersion: 3.5.1
+InstallerSwitches:
+  Silent: /Q
```
Comment (OfficialEsco) [inline manifests/m/Mintty/WSLtty/3.5.1/Mintty.WSLtty.installer.yaml:7]: > I can't see what's causing the validation checks to fail.

```suggestion
InstallerSwitches:
  Silent: /Q
  SilentWithProgress: /Q
```

**Comment by mintty** [comment]
> Perhaps this might be something to report to the developers of wsltty? cc @mintty

Please note that there is also a repository https://github.com/mintty/wsltty.appx, which would initially support manual installation of an appx file. But it never worked out to become installable via the usual means and the project is now very outdated.
I'd appreciate to get wsltty available as an installable app but I have no idea what the current problems are.

**Comment by me-and** [comment]
@mintty I think the issue here is with the `/Q` switch.

If you run `wsltty-3.5.1-x86_64-install.exe /?`, you get a help dialog that suggests command line options including "/Q -- Quiet modes for the package," so I added `/Q` as the way for winget to run the installer quietly. By default, winget runs all installers in quiet mode, so that switch gets specified. But when I run the installer with that switch, I see the same error @OfficialEsco and @ItzLevvie saw: "Error creating process \<\<None\>\>. Reason: The system cannot find the file specified."

(The reason everything worked fine for me when I was testing, I think, is that I tested – as I nomally do – with `winget install --interactive ...`.)

So for the installer to be available as part of winget, there needs to be a quiet/silent/non-interactive mode for the WSLtty installer that actually works. I've not looked at how the WSLtty installer is generated, so I have no idea what work is necessary for that. Let me know if it'd be useful to raise an issue over at mintty/wsltty, though!

**Comment by mintty** [comment]
I used the most basic and easily usable installer I could find a while ago, the quite antique iexpress which seems to be something intended to preconfigure internet explorer somehow. I would have liked to rather use a modern installer generator but was appalled by their complexity and the need to spend significant research into them before getting a result, so that's the story; iexpress itself has a /Q option but it doesn't make the resulting installer q-able.
Can you use a .bat file as an installer? A simple wrapper that accepts and ignores /Q?

**Comment by mintty** [comment]
There is an initial "Install wsltty?" confirmation dialog, then a command window which logs all the extraction and setup commands, and finally a confirmation popup "Installed". The middle part, the command line log, can actually be suppressed. Would that help?

**Comment by mintty** [comment]
Actually it is also possible to skip the messages. There will be an instant flicker of an initial window, for 0.1 s.
I can provide such an installer for testing if you like.

**Comment by mintty** [comment]
> There will be an instant flicker of an initial window, for 0.1 s.

Managed to skip that too, so...

**Comment by OfficialEsco** [comment]
Any installer submitted to WinGet need to be from the Official publisher, so if you want wsltty on WinGet please do =)
Will the new installer require no switches?

**Comment by mintty** [comment]
Yes, no switches:
[wsltty-3.5.1-quiet-installer.zip](https://github.com/microsoft/winget-pkgs/files/7857608/wsltty-3.5.1-quiet-installer.zip)

**Comment by OfficialEsco** [comment]
> Yes, no switches:
> [wsltty-3.5.1-quiet-installer.zip](https://github.com/microsoft/winget-pkgs/files/7857608/wsltty-3.5.1-quiet-installer.zip)

heh
VirtualBoxVM_XPFnl6uDaj
VirtualBoxVM_i61pWciy6C

**Comment by mintty** [comment]
There have been repeated reports of virus suspicions over time in the wsltty issue tracker, all different.
It's likely because Windows does not trust its own legacy installer, or the old installer does not provide any security information, or whatever. I do not know how to sort that out. In case you have doubts, you are welcome to follow the installer creation procedure yourself from the repository, based on a minimal cygwin installation.

**Comment by OfficialEsco** [comment]
https://github.com/mintty/wsltty/releases/download/3.5.1/wsltty-3.5.1-x86_64-install.exe seems to not be flagged tho, however if the new method is flagged for any of the AV's it will not pass one of the validation.

**Comment by mintty** [comment]
I had also stripped some stale settings on that occasion which I've now reverted, so may this is a less different version:
[wsltty-3.5.1-quiet-installer2.zip](https://github.com/microsoft/winget-pkgs/files/7858006/wsltty-3.5.1-quiet-installer2.zip)

**Comment by me-and** [comment]
IIRC winget doesn't currently support extracting files from zip archives,
so to test this we'd need to have it uploaded somewhere as a bare .exe
installer.

I'm guessing the restriction on only using official installers might mean
it also needs to be clearly attached to an official source -- so probably
on the wsltty repo, or somewhere on https://mintty.github.io/ -- but I
assume that we could use an .exe attached to this ticket to at least prove
winget is capable of downloading and installing the package.

**Comment by mintty** [comment]
> prove winget is capable of ...

That's exactly the purpose of this upload of course. Just unpack the zip for a test case; github does not accept exe uploads.

**Comment by OfficialEsco** [comment]
>  I've specifying Silent: null (which my reading of https://aka.ms/winget-manifest.installer.1.0.0.schema.json implies is correct), Silent: /Q and Silent:, all of which produced the same behaviour.

So this is why people use `Silent: null` or `Silent: false`/`true`?
the InstallerSwitches field is not required, so it can be removed, however it will throw a validation Warning
a Workaround to the warning is
```
InstallerSwitches
  Silent: ' '
  SilentWithProgress: ' '
```
I believe null refers to NullOrWhiteSpace

I executed the second installer yesterday, and it didn't get blocked, VirusTotal showed 4 Flags, one from Microsoft marked as Trojan which is weird since it didn't get blocked
A shortcut appeared on the Desktop but no `A`dd or `R`emove `P`rograms entries which might be intended behavior

**Comment by mintty** [comment]
> A shortcut appeared on the Desktop but no Add or Remove Programs entries which might be intended behavior

A successful installation should give you one desktop shortcut, and N+1 Start Menu shorcuts (one for each WSL distribution available at the time of installation, plus one for the default WSL distribution), plus a WSLtty folder in the Start Menu with additional shorcuts and configuration scripts.

**Comment by me-and** [comment]
So it sounds like that has confirmed the new silent installer works as
expected! @mintty are you able to upload that (presumably in both 32- and
64-bit flavours) to the main WSLtty repository? Once the .exe installer is
available there, it should be possible to update this PR with the new
links, and make WSLtty available through winget.

**Comment by mintty** [comment]
I've skipped mintty 3.5.2 for wsltty, and I'm about to release mintty 3.5.3, possibly also with an update of the wslbridge2 gateway. Should I wait or would you still like an upload of the 3.5.1 release?

**Comment by me-and** [comment]
It sounds like waiting makes sense if it's going to happen imminently :)

**Review by ItzLevvie — DISMISSED** [review]

**Comment by me-and** [comment]
I've just converted this PR to a draft: notwithstanding it having moderator approval already, I don't think it's ready to merge until the new wsltty release, with the silent installer, is available to download from the wsltty GitHub repo (or some other official source, but definitely not the spare system I happen to be able to serve files from!)

**Comment by mintty** [comment]
Released wsltty 3.5.3: https://github.com/mintty/wsltty/releases/tag/3.5.3

**Review by OfficialEsco — APPROVED** [review]

**Comment by mintty** [comment]
Thank you, I appreciate this very much.
How should future releases be handled? Should I notify them here?

**Comment by OfficialEsco** [comment]
* If you want to maintain it your self you can make some automation or use wingetcreate.
* You can request the package to be added to https://github.com/vedantmgoyal2009/winget-pkgs-automation (Will most likely be added some time eitherway)
* Or since you're a good boy that release to GitHub, we with RSS feeds will eventually catch the update until MS adapts the automation

**Comment by mintty** [comment]
Released wsltty 3.6.0.

**Comment by OfficialEsco** [comment]
> Released wsltty 3.6.0.

Microsoft Defender dosn't really like you tho
```
PS M:\> winget install -m m\Mintty\WSLtty\3.6.0\
Found WSLtty [Mintty.WSLtty] Version 3.6.0
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://github.com/mintty/wsltty/releases/download/3.6.0/wsltty-3.6.0-x86_64-install-quiet.exe
  ██████████████████████████████  2.40 MB / 2.40 MB
Successfully verified installer hash
An anti-virus product reports an infection in the installer
```
VirtualBoxVM_oQmfDuIogV

**Pull request merged**